src/material/schematics/ng-generate/mdc-migration/rules/style-migrator.ts

```diff
@@ -91,7 +94,9 @@ export abstract class StyleMigrator {
    * @returns `true` if the given Rule uses a legacy selector of this component.
    */
   isLegacySelector(rule: postcss.Rule): boolean {
-    return this.classChanges.some(change => rule.selector.includes(change.old));
+    // Since a legacy class can also have the deprecated prefix, we also
+    // check that a match isn't actually a longer deprecrated class.
```

Comment:
```suggestion
    // check that a match isn't actually a longer deprecated class.
```

Is it guaranteed that the selectors are put into a single line? what happens if a multi-line selector is used? We may need to add a test for this, and likely need to handle `\s` (which could be any type of whitespace?)